### PR TITLE
[Data studios] [enterprise 24.1] update docker compose with connect pieces

### PR DESCRIPTION
**Pull request by weronikasosnowskaseqera**
Closes https://github.com/seqeralabs/tower-connect/issues/120

This PR adds configuration and documentation for `docker-compose` with data studios (connect proxy and server). 

To test It you need to have setup to expose tower, connect and studio domain. To do this with ngrok you need an account.

### CODE REVIEW

**Comment by justinegeffen** [comment]
Thanks, @weronikasosnowskaseqera! I've added some comments. LMK when I can do a second pass.

@endre-seqera, if you're able to review/approve tomorrow that'd be great. We need to merge this into the release branch ASAP. Thank you!

File: platform_versioned_docs/version-24.1.0/enterprise/data-studios.mdx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    volumes:
+      - $PWD/data-studios-rsa.pem:/data-studios-rsa.pem
+```
+
+## Kubernetes
+TBD
+
+## DNS setup
+Proxy domain (`CONNECT_PROXY_URL`) has to be a subdomain of main platform domain (example: if platform domain is
+`tower.io`, then proxy domain should be `connect.tower.io`).
```
Comment (t0randr) [inline platform_versioned_docs/version-24.1.0/enterprise/data-studios.mdx:71]: The domain can be different from `connect`:  any valid subdomain will work.

File: platform_versioned_docs/version-24.1.0/enterprise/data-studios.mdx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+Data studios require a wildcard DNS record that will match request for non-existent subdomains.
+It is specified by a `*` as part of the domain name, example: `*.tower.io`. It should be pointed to the port 9090.
```
Comment (t0randr) [inline platform_versioned_docs/version-24.1.0/enterprise/data-studios.mdx:73]: They also need a valid wildcard ssl certificate to cover the third level domain.

We might want to suggest to use only a third level domain if possible, so that they can use a cheaper certificate.

Comment (weronikasosnowskaseqera) [reply platform_versioned_docs/version-24.1.0/enterprise/data-studios.mdx:73]: Great catches! Thanks for the comments, I will change that now.

File: platform_versioned_docs/version-24.1.0/enterprise/data-studios.mdx [hunk 3]
```diff
@@ -0,0 +1,79 @@
+---
+title: "Data studios deployment"
+description: Deploy Seqera Platform with Data studios
+date: "12 Jun 2024"
+tags: [docker, compose, kubernetes, data, studios, deployment]
+---
+
+This guide assumes that all prerequisites have been met. Visit the corresponding **Prerequisites** page for your infrastructure provider.
```
[the remark this answers is not in the document]
Comment (weronikasosnowskaseqera) [reply platform_versioned_docs/version-24.1.0/enterprise/data-studios.mdx:8]: I will rephrase It to the docker-compose specific case, but FYI there are more general prerequisites 
 mentioned in docs tasks https://github.com/seqeralabs/platform/issues/6979

File: platform_versioned_docs/version-24.1.0/enterprise/data-studios.mdx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+description: Deploy Seqera Platform with Data studios
+date: "12 Jun 2024"
+tags: [docker, compose, kubernetes, data, studios, deployment]
+---
+
+## Docker compose
+
+This guide assumes that all services will be run in the same container as the rest of the platform services.
+
+To run data studios in docker compose setup uncomment following values in the `_templates/docker/tower.env` file:
```
Comment (justinegeffen) [inline platform_versioned_docs/version-24.1.0/enterprise/data-studios.mdx:12]:
```suggestion
To run Data Studios in `docker compose` setup uncomment following values in the `_templates/docker/tower.env` file:
```

File: platform_versioned_docs/version-24.1.0/enterprise/data-studios.mdx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+```dotenv
+# Data studios settings
```
Comment (justinegeffen) [inline platform_versioned_docs/version-24.1.0/enterprise/data-studios.mdx:15]:
```suggestion
# Data Studios settings
```

File: platform_versioned_docs/version-24.1.0/enterprise/data-studios.mdx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+Example of setup:
+```
+*.tower.io -> 9090
+tower.io -> 8000
+```
+
+Additionally, a valid wildcard ssl certificate to cover the third level domain is required.
+
+!!! warning It is recommended to use only a third level domain if possible, as the cost may increase with the number of subdomains.
```
Comment (justinegeffen) [inline platform_versioned_docs/version-24.1.0/enterprise/data-studios.mdx:82]:
```suggestion
:::note
We recommend using only a third-level domain if possible, as costs may increase with the number of subdomains.
:::
```

**Review by justinegeffen — APPROVED** [review]
Minor comments but this LGTM! Approving.
